### PR TITLE
Experimental tool: Pyblish debug stepper

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
Add pyblish debug stepper to experimental tools 
For more info about it go to [Pyblish Debug Stepper | Forums](https://community.ynput.io/t/pyblish-plugins-debugging/1447#h-2-pyblish-debug-stepper-2)

## Additional Info
Mostly quoted from @iLLiCiTiT 
Experimental tools are/were meant for artists. As temporary location for tools that will eventually go to standard menu. (no hard opinion about it) 
It is still a matter of debate whether the tool should be located in Experimental tools or not.

## Demo
Here's a quick demo :wink: 

https://github.com/ynput/ayon-core/assets/20871534/2742aa18-8f09-4fec-a97f-a16f25f33b1f

## Testing notes:
As shown in the demo

### CODE REVIEW

File: client/ayon_core/tools/experimental_tools/tools_def.py [hunk 1]
```diff
@@ -162,6 +169,13 @@ def refresh_availability(self):
             local_settings.get(LOCAL_EXPERIMENTAL_KEY)
         ) or {}
 
+        # Enable the following tools by default.
+        # Because they will always be disabled due
+        #   to the fact their settings don't exist.
+        experimental_settings.update({
+            "DebugExprimentalTool": True,
```
Comment (BigRoy) [inline client/ayon_core/tools/experimental_tools/tools_def.py:176]: Typo?
```suggestion
            "DebugExperimentalTool": True,
```

Comment (MustafaJafar) [reply client/ayon_core/tools/experimental_tools/tools_def.py]: Good catch! 
Fixed here https://github.com/ynput/ayon-core/pull/753/commits/f5006273ae69f6cb6404522488ae3141ed376200

**Comment by MustafaJafar** [comment]
I'm so happy that this PR has no conflicts so far 😄

**Comment by tanh609** [comment]
Awesome tool 💯. Thank you so much. 

It would greatly help if we can have a shortcut button for the "Step" button. It feel like cookie clicker when there are a lot of steps.

**Comment by BigRoy** [comment]
@MustafaJafar I pushed some changes - support for PySide6, some cosmetics and adding a simple filter field for the keys.

**Review by BigRoy — APPROVED** [review]
I'm a fan. :)

Still tons we can improve feature wise but it's a great tool to have handy when debugging.

**Comment by MustafaJafar** [comment]
That filter is cool! 
image

**Comment by BigRoy** [comment]
> It would greatly help if we can have a shortcut button for the "Step" button. It feel like cookie clicker when there are a lot of steps.

Any huge preference? Does pressing `step` once and then clicking "enter" or "space" suffice? (I assume that button becomes 'active'?)

**Pull request merged**